libsrc/leddevice/dev_net/LedDeviceWled.cpp

```diff
@@ -0,0 +1,301 @@
+// Local-Hyperion includes
+#include "LedDeviceWled.h"
+
+// ssdp discover
+#include <ssdp/SSDPDiscover.h>
+
+// Configuration settings
+static const char CONFIG_ADDRESS[] = "host";
+
+// UDP elements
+const quint16 STREAM_DEFAULT_PORT = 19446;
+
+// WLED JSON-API elements
+static const int API_DEFAULT_PORT = -1; //Use default port per communication scheme
+
+static const char API_BASE_PATH[] = "/json/";
+static const char API_PATH_INFO[] = "info";
+static const char API_PATH_STATE[] = "state";
+
+// List of State Information
+static const char STATE_ON[] = "on";
+static const char STATE_VALUE_TRUE[] = "true";
+static const char STATE_VALUE_FALSE[] = "false";
+
+// WLED ssdp services
+// TODO: WLED - Update ssdp discovery parameters when available
+static const char SSDP_ID[] = "ssdp:all";
+static const char SSDP_FILTER[] = "(.*)";
+static const char SSDP_FILTER_HEADER[] = "ST";
+
+LedDeviceWled::LedDeviceWled(const QJsonObject &deviceConfig)
+	: ProviderUdp()
+	  ,_restApi(nullptr)
+	  ,_apiPort(API_DEFAULT_PORT)
+{
+	_devConfig = deviceConfig;
+	_isDeviceReady = false;
+
+	_activeDeviceType = deviceConfig["type"].toString("UNSPECIFIED").toLower();
+}
+
+LedDeviceWled::~LedDeviceWled()
+{
+	if ( _restApi != nullptr )
+	{
+		delete _restApi;
+		_restApi = nullptr;
+	}
+}
+
+LedDevice* LedDeviceWled::construct(const QJsonObject &deviceConfig)
+{
+	return new LedDeviceWled(deviceConfig);
+}
+
+bool LedDeviceWled::init(const QJsonObject &deviceConfig)
+{
+	Debug(_log, "");
+	bool isInitOK = false;
+
+	// Initialise LedDevice sub-class, ProviderUdp::init will be executed later, if connectivity is defined
+	if ( LedDevice::init(deviceConfig) )
+	{
+		// Initialise LedDevice configuration and execution environment
+		uint configuredLedCount = this->getLedCount();
+		Debug(_log, "DeviceType   : %s", QSTRING_CSTR( this->getActiveDeviceType() ));
+		Debug(_log, "LedCount     : %u", configuredLedCount);
+		Debug(_log, "ColorOrder   : %s", QSTRING_CSTR( this->getColorOrder() ));
+		Debug(_log, "LatchTime    : %d", this->getLatchTime());
+
+		//Set hostname as per configuration
+		QString address = deviceConfig[ CONFIG_ADDRESS ].toString();
+
+		//If host not configured the init fails
+		if ( address.isEmpty() )
+		{
+			this->setInError("No target hostname nor IP defined");
+			return false;
+		}
+		else
+		{
+			#if (QT_VERSION >= QT_VERSION_CHECK(5, 14, 0))
```

Comment:
Maybe we should encapsulate this in an internal string helper library instead of repeating those if elses everytime we need a split

Reply:
Good idea. You would like to have a go and put it under libsrc/utils/QStringutils.h ?

Reply:
Maybe later :)

Reply:
I added a QStringUtil.h with an inline function.
Please have a look when i commit later....